camp/settings.py

```diff
@@ -212,113 +212,46 @@
-    "visibility": False,
-    "wrapDateLine": True,
-    "fixed": True,
-    "group":"background"
-# }, {
-#     "source": {"ptype": "gxp_olsource"},
-#     "type": "OpenLayers.Layer.XYZ",
-#     "title": "MapBox Satellite Streets",
-#     "args": ["MapBox Satellite Streets", "http://api.mapbox.com/styles/v1/mapbox/satellite-streets-v9/tiles/${z}/${x}/${y}?access_token="+MAPBOX_ACCESS_TOKEN],
-#     "name": "background",
-#     "attribution": "&copy; <a href='https://www.mapbox.com/about/maps/'>Mapbox</a> &copy; <a href='http://www.openstreetmap.org/copyright'>OpenStreetMap</a> <a href='https://www.mapbox.com/feedback/' target='_blank'>Improve this map</a>",
-#     "visibility": False,
-#     "wrapDateLine": True,
-#     "fixed": True,
-#     "group":"background"
-# }, {
-#     "source": {"ptype": "gxp_olsource"},
-#     "type": "OpenLayers.Layer.XYZ",
-#     "title": "MapBox Streets",
-#     "args": ["MapBox Streets", "http://api.mapbox.com/styles/v1/mapbox/streets-v9/tiles/${z}/${x}/${y}?access_token="+MAPBOX_ACCESS_TOKEN],
-#     "name": "background",
-#     "attribution": "&copy; <a href='https://www.mapbox.com/about/maps/'>Mapbox</a> &copy; <a href='http://www.openstreetmap.org/copyright'>OpenStreetMap</a> <a href='https://www.mapbox.com/feedback/' target='_blank'>Improve this map</a>",
-#     "visibility": False,
-#     "wrapDateLine": True,
-#     "fixed": True,
-#     "group":"background"
-}, {
-    "source": {"ptype": "gxp_osmsource"},
-    "type": "OpenLayers.Layer.OSM",
-    "title": "OpenStreetMap",
-    "name": "mapnik",
-    "attribution": "&copy; <a href='http://osm.org/copyright'>OpenStreetMap</a> contributors",
-    "visibility": True,
-    "wrapDateLine": True,
-    "fixed": True,
-    "group": "background"
-}, {
-    "source": {
-        "ptype": "gxp_tianditusource"
-    },
-    "group": "background",
-    "name": "TIANDITUROAD",
-    "visibility": True,
-    "fixed": True,
-}, {
-    "source": {
-        "ptype": "gxp_tianditusource",
-    },
-    "group": "background",
-    "name": "TIANDITUIMAGE",
-    "visibility": False,
-    "fixed": True,
-}, {
-    "source": {
-        "ptype": "gxp_tianditusource"
-    },
-    "group": "background",
-    "name": "TIANDITUTERRAIN",
-    "visibility": False,
-    "fixed": True,
-}, {
-    "source": {
-        "ptype": "gxp_tianditusource"
-    },
-    "group": "background",
-    "name": "TIANDITUANNOTATION",
-    "visibility": True,
-    "fixed": True,
-}]
+if TIANDITU_API_KEY:
+    MAP_BASELAYERS += [{
+        "source": {
+            "ptype": "gxp_tianditusource",
+            "key":TIANDITU_API_KEY
+        },
+        "group": "background",
+        "name": "TIANDITUROAD",
+        "visibility": True,
+        "fixed": True,
+    }, {
+        "source": {
+            "ptype": "gxp_tianditusource",,
```

Comment:
there is one extra comma here which is causing a syntax error

Reply:
Sorry, it's my fault.